frontend/src/pages/Admin/Users/index.tsx

```diff
@@ -1,21 +1,154 @@
 import React, { FC } from "react";
-import { LinearProgress, Typography } from "@mui/material";
+import { Typography } from "@mui/material";
 import { useFormatTableData } from "../../../hooks/";
+import { useFormControls } from "hooks/useFormControls";
+import { useFormSubmit } from "hooks/useFormSubmit";
 import { Table } from "../../../components";
+import { apiAxios } from "utils";
+import { useQuery } from "react-query";
+import { Renderer } from "components/Renderer";
+import { ReadForm } from "components/ReadForm";
+import { EditForm } from "components/EditForm";
+import { Box, Button } from "@mui/material";
+import { CreateForm } from "components/CreateForm";
+import { GDXModal } from "components/GDXModal";
+import { editFields, readFields } from "./fields";
 
 export const Users: FC = () => {
-  const { data, isLoading } = useFormatTableData({ tableName: "users", apiEndPoint: "users" });
+  const {
+    handleEditMode,
+    handleOpen,
+    handleClose,
+    handleCurrentRowData,
+    handleFormType,
+    formType,
+    open,
+    editMode,
+    currentRowData,
+  } = useFormControls();
+
+  const { data, isLoading } = useFormatTableData({
+    tableName: "users",
+    apiEndPoint: `users`,
+    handleClick: handleOpen,
+  });
+
+  const { handlePost, handleUpdate, Notification } = useFormSubmit();
+  /**
+   * Gets the user information for a specific id.
+   *
+   * @returns {null|object}
+   */
+  const getUsers = async () => {
+    let data = null;
+    if (currentRowData?.id) {
+      const users = await apiAxios().get(`/users/${currentRowData?.id}`);
+      data = users.data.data;
+    }
+    return data;
+  };
+
+  // Queries
+  /* eslint "no-warning-comments": [1, { "terms": ["todo", "fixme"] }] */
+  // todo: Define a good type. "Any" type temporarily permitted.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const userQuery: any = useQuery(`users - ${currentRowData?.id}`, getUsers, {
```

Comment:
I had to cancel my first comment because I thought you called this "useQuery" and I was going to say something but then I had a closer glance..  Tricky